computer_vision/classification/alexnet/run.py

```diff
@@ -1,10 +1,20 @@
 # SPDX-License-Identifier: Apache-2.0
 # Copyright (c) 2022, Ampere Computing LLC
 
+import os
+import sys
 import argparse
 
 import torch
 import torchvision
+from pathlib import Path
+
+try:
+    from utils.cv.imagenet import ImageNet
+except ModuleNotFoundError as e:
+    pythonpath = os.path.join(Path(os.getcwd(), __file__))
```

Comment:
__file__ gives you what you need already. No need for os.getcwd(). Also, why not just use an already working solution from mzdev?

Reply:
https://github.com/AmpereComputingAI/model_zoo_dev/blob/main/downloads/utils.py

Reply:
Lets create a util under utils setting the python_path with an unified approach that could be imported by every runner.

Reply:
__file__ returns the path to the file which is run from your working directory. If the user enters folder "alexnet", and runs the script from there "__file__" will return "run.py", the way I did this it always returns an abosolute path to the file which is run. From this path I can then set the pythonpath in the run script.

If the pythonpath is not set then such a function in utils will not be importable and thus will not work, or am I missing something?

Reply:
You are of course right on inability to use utils without setting PYTHON_PATH first.

But __file__ seems to return absolute path of file no matter where you run it:

> jan@g-workstation:~/Downloads$ cat test.py 
> print(__file__)
> jan@g-workstation:~/Downloads$ python3 test.py 
> **/home/jan/Downloads/test.py**
> jan@g-workstation:~/Downloads$ cd ../..
> jan@g-workstation:/home$ python3 jan/Downloads/test.py
> **/home/jan/Downloads/test.py**

Reply:
So, os.getcwd() is redundant.

Reply:
strange, I get a different behavior, nonetheless os.getcwd() was redundant, I solved it in a different way